src/librustc_mir/interpret/const_eval.rs

```diff
@@ -123,20 +130,23 @@ pub fn value_to_const_value<'a, 'tcx>(
             }
         }
         _ => {
-            if let Value::ByRef(ptr, align) = val {
+            if let Value::ByRef(ptr, _) = val {
                 let ptr = ptr.primval.to_ptr().unwrap();
                 assert_eq!(ptr.offset, 0);
-                let alloc = tcx.interpret_interner
-                               .get_alloc(ptr.alloc_id)
-                               .expect("miri allocation never successfully created");
-                assert_eq!(align, alloc.align);
+                let alloc = ecx.memory.get(ptr.alloc_id).unwrap();
```

Comment:
please use expect so we can find it quickly if it fails (even if the issue reported has no backtrace)

Reply:
I've changed this to report any miri errors then panic.